Sources/NIOSSL/ObjectIdentifier.swift

```diff
@@ -53,65 +63,89 @@ public struct NIOSSLObjectIdentifier: Hashable {
     /// Creates a Object Identifier (OID) from its textual dotted representation (e.g. `1.2.3`)
     /// - Parameter string: textual dotted representation of an OID
     public init?(_ string: String) {
-        guard let storage = Storage(string) else {
+        let result = string.withCString { string in
+            // If no_name (the last parameter of CNIOBoringSSL_OBJ_txt2obj) is 0 then long names and
+            // short names will be interpreted as well as numerical forms.
+            // If no_name is 1 only the numerical form is acceptable.
+            // source: https://www.openssl.org/docs/manmaster/man3/OBJ_txt2obj.html
+            CNIOBoringSSL_OBJ_txt2obj(string, 1)
+        }
+        guard let reference = result else {
             return nil
         }
-        self.storage = storage
+        self.storage = .init(takeOwnershipOf: reference)
     }
     
     /// Creates an Object Identifier (OID) from an OpenSSL reference.
     /// - Note: initialising an ``NIOSSLObjectIdentifier`` takes ownership of the reference and will free it after the reference count drops to zero
     /// - Parameter reference: reference to a valid OpenSSL OID aka OBJ
-    internal init(takingOwnershipOf reference: OpaquePointer) {
-        self.storage = Storage(takingOwnershipOf: reference)
+    internal init(takingOwnershipOf reference: OpaquePointer!) {
+        self.storage = .init(takeOwnershipOf: reference)
     }
-}
-
-extension NIOSSLObjectIdentifier.Storage: Equatable {
-    fileprivate static func == (lhs: NIOSSLObjectIdentifier.Storage, rhs: NIOSSLObjectIdentifier.Storage) -> Bool {
-        CNIOBoringSSL_OBJ_cmp(lhs.reference, rhs.reference) == 0
+    
+    /// Creates an Object Identifier (OID) from an OpenSSL reference.
+    /// - Note: initialising an ``NIOSSLObjectIdentifier`` does **not** take ownership of the memory. Instead ``NIOSSLObjectIdentifier`` keeps a reference to the owning object which it will retain for the lifetime of itself.
+    /// - Parameters
+    ///   - reference: reference to a valid OpenSSL OID aka OBJ
+    ///   - owner: owner of the memory `reference` is pointing to which it will retain.
+    internal init(takeUnowned reference: OpaquePointer!, owner: AnyObject) {
```

Comment:
I think a better set of labels is 

```suggestion
    internal init(borrowing reference: OpaquePointer!, owner: AnyObject) {
```